src/instantiate.rs

```diff
@@ -19,6 +22,8 @@ pub fn instantiate(
     info: MessageInfo,
     msg: InstantiateMsg,
 ) -> ContractResponse {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
```

Comment:
So this sets the current version of the contract, then if you migrate one it updates the contract version but what is the mechanism that is able to load the old structure into the new structure? Is that all just the magic of the cw20 lib?

Reply:
assuming structure means state, the state in storage is left untouched, however the unit test below shows we are able to deserialize with the new state dto just fine with pre existing target attributes in the raw storage